### ISSUE TITLE
Allow interface function names in `SNAKE_CASE` when they represent constant values

### ISSUE THREAD

**Issue by smol-ninja** [protofire/solhint#699]
### Problem

Consider the following example:

```solidity
interface IA {
    /// @dev This is a constant state variable.
    function START_TIME() external view returns (uint256); 
}

contract A is IA {
    /// @inheritdoc IA
    uint256 public constant override START_TIME = 1;
}
```

The `func-name-mixedcase` rule currently throws an error for the above example. However, this is a valid naming convention when the function returns a constant value. The same for immutables.

### Solution

I believe there are two possible approaches to address this:

1. **Conditional exemption**: Ignore `SNAKE_CASE` function names in interface files when they represent constant values (e.g., when documented as constants or immutables in natspecs).

2. **New rule**: Add a complementary rule like `const-func-name-snakecase` that enforces constant/immutable function names to be in `SNAKE_CASE`.

**Comment by dbale-altoros** [comment]
check
https://github.com/protofire/solhint/pull/705

**Comment by smol-ninja** [comment]
@dbale-altoros your approach sounds good to me. The above conditions should be enough except for the "return type". 

For instance:

```solidity
// This is also valid constant.
function TOKEN() external view returns (IERC20);
```

Similarly, there could be other types wrapped over elementary types:

```solidity
// UD60x18 is type uint256
function VESTING_CLIFF_UNLOCK_PERCENTAGE() external view returns (UD60x18);
```

So, it would be nice to allow these as well.

**Comment by dbale-altoros** [comment]
@smol-ninja good one
I'll review if there are other ones

**Comment by dbale-altoros** [comment]
@smol-ninja 

Be declared in an interface.
Have external view visibility and mutability.
Have zero parameters.
Have exactly one return value.
The return type must be one of the following:
    An elementary type (uint*, int*, bool, address, string, bytes, bytesN).
    An enum.
    A user-defined value type (UDVT) based on an elementary type (e.g. UD60x18).
    A contract/interface type (e.g. IERC20).

Not allowed: arrays, tuples, structs, or any return type with a storageLocation (memory/calldata).
